frontend/src/metabase/visualizations/shared/settings/column.ts

```diff
@@ -1,6 +1,13 @@
 import { isCurrency, isPercentage } from "metabase-lib/v1/types/utils/isa";
 import type { ColumnSettings, DatasetColumn } from "metabase-types/api";
 
+function hasDataSizeName(column: DatasetColumn): boolean {
```

Comment:
We should not hardcode column type detection like this on the frontend, this is done via semantic type system like below in `getDefaultNumberStyle` function via utils like `isCurrency` or `isPercentage` that check for semantic types. Since based on @ranquild content we don't want to add a new semantic type here, we need to remove this auto-detection code because it is a wrong place to do it

Reply:
Ok, so just to be clear, are you therefore saying there should now be no automatic detection of columns that are clearly a data size type and only rely on a user configuring that, or have I misunderstood your comment here?